### PR TITLE
Add parameter for connection demodulator with selected oscillator

### PR DESCRIPTION
**Pull request by qSaevar**
Changes proposed in this pull request:
-  Add parameter to UHFLI driver to select oscillator per demodulator

@WilliamHPNielsen

### CODE REVIEW

File: qcodes/instrument_drivers/ZI/ZIUHFLI.py [hunk 1]
```diff
@@ -710,6 +710,17 @@ def __init__(self, name: str, device_ID: str, **kwargs) -> None:
...
 
+            self.add_parameter('demod{}_oscillator'.format(oscs),
+                               label='Selected oscillator {}'.format(oscs),
+                               docstring="Connects the demodulator with the "
+                                         "supplied oscillator.",
+                               get_cmd=partial(self._getter, 'demods',
+                                               oscs-1, 0, 'oscselet'),
+                               set_cmd=partial(self._setter, 'demods',
+                                               oscs-1, 0, 'oscselect'),
+                               val_mapping={i+1: i for i in range(8)},
```
Comment (WilliamHPNielsen) [inline qcodes/instrument_drivers/ZI/ZIUHFLI.py]: Should this range also be `number_of_oscillators` ?

Comment (qSaevar) [reply qcodes/instrument_drivers/ZI/ZIUHFLI.py]: Yes, it should.

Comment (qSaevar) [reply qcodes/instrument_drivers/ZI/ZIUHFLI.py]: same for vals, it's fixed now

File: qcodes/instrument_drivers/ZI/ZIUHFLI.py [hunk 2]
```diff
@@ -718,8 +718,9 @@ def __init__(self, name: str, device_ID: str, **kwargs) -> None:
                                                oscs-1, 0, 'oscselet'),
                                set_cmd=partial(self._setter, 'demods',
                                                oscs-1, 0, 'oscselect'),
-                               val_mapping={i+1: i for i in range(8)},
-                               vals=vals.Ints(1, 8))
+                               val_mapping={i + 1: i for i in
+                                            range(number_of_oscillators)},
+                               vals=vals.Ints(1, number_of_oscillators))
```
Comment (WilliamHPNielsen) [inline qcodes/instrument_drivers/ZI/ZIUHFLI.py]: This is actually redundant, the `val_mapping` creates a default validator which is just an `Enum` validator of the dict keys.

Comment (qSaevar) [reply qcodes/instrument_drivers/ZI/ZIUHFLI.py]: I removed the vals

**Comment by qSaevar** [comment]
@WilliamHPNielsen Great thanks. I have only tested this on a device with MF enabled.

**Comment by WilliamHPNielsen** [comment]
@qSaevar, that's what I thought. And I have one without MF, so that _could_ be good to test, but I'll risk it, just this once.

**Pull request merged**